.github/workflows/pre_merge.yml

```diff
@@ -52,19 +48,37 @@ jobs:
     concurrency:
       group: ${{ github.workflow }}-Unit-${{ github.event.pull_request.number || github.ref }}-${{ matrix.tox-env }}
       cancel-in-progress: true
+    uses: ./.github/workflows/run_tests_in_tox_custom.yml
+    with:
+      python-version: ${{ matrix.python-version }}
+      toxenv-pyver: ${{ matrix.tox-env }}
+      toxenv-task: all
+      tests-dir: tests/unit
+      timeout-minutes: 120
+      upload-artifact: true
+      artifact-prefix: "unit-test-results"
+      runs-on: "['otx-gpu-v100-1']"
+  Coverage-Test:
+    needs: Code-Quality-Checks
+    concurrency:
+      group: ${{ github.workflow }}-Coverage-${{ github.event.pull_request.number || github.ref }}}
+      cancel-in-progress: true
+    runs-on: ["otx-gpu-v100-1"]
+    container:
+      image: 219678651685.dkr.ecr.eu-central-1.amazonaws.com/ote-ci:pr-194-1334bef936df3b06d28bc3d1bd90e7248359b99e
+      options: --runtime=nvidia --env-file=/home/runner/.nvidia.env --shm-size=24g
     steps:
       - name: Checkout repository
         uses: actions/checkout@v3
       - name: Install Python
         uses: actions/setup-python@v4
         with:
-          python-version: ${{ matrix.python-version }}
+          python-version: "3.8"
```

Comment:
Didn't you delete Python 3.8?

Reply:
this is just for the coverage run.

Reply:
Python 3.8 was put into EOL last year, shouldn't this be made to 3.10 as well?